subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuildRegistry.java

```diff
@@ -240,6 +247,36 @@ private BuildIdentifier idFor(String buildName) {
     }
 
     private Path assignPath(BuildState owner, String name, File dir) {
+        // Get the closest ancestor build of the build directory which we are currently adding
+        Optional<Map.Entry<File, IncludedBuildState>> parentBuild = includedBuildsByRootDir.entrySet().stream()
```

Comment:
It would be good to remove the special casing for included build and buildSrc, by maintaining a map from root dir -> build for all builds. Then we should be able to use the same assignment strategy for all builds.

Reply:
Good idea, done!